### PR TITLE
[rtsan] Add include guards around posix interceptors, tests

### PR DESCRIPTION
**Pull request by cjappl**
This is another in preparation for windows exploration, trying to get all the sub-pieces in place

### CODE REVIEW

**Comment by cjappl** [comment]
CC @jatinchowdhury18

**Comment by davidtrevelyan** [comment]
Is there strong motivation for doing this in the source rather than in the CMake configuration?

**Comment by cjappl** [comment]
> Is there strong motivation for doing this in the source rather than in the CMake configuration?

There seems to be strong precedent to do it via ifdefs. Take for instance the platform specific malloc files in asan:

https://github.com/llvm/llvm-project/blob/54c93aabec965469fe7db1f4391a190e3d640feb/compiler-rt/lib/asan/asan_malloc_mac.cpp#L15

https://github.com/llvm/llvm-project/blob/54c93aabec965469fe7db1f4391a190e3d640feb/compiler-rt/lib/asan/asan_malloc_win.cpp#L16

https://github.com/llvm/llvm-project/blob/54c93aabec965469fe7db1f4391a190e3d640feb/compiler-rt/lib/asan/asan_malloc_linux.cpp#L17

I'm not opposed to doing this in CMake (and it would be my first gut reaction in a project I controlled) but this seems to be the way I've seen around the codebase.

**Comment by davidtrevelyan** [comment]
> > Is there strong motivation for doing this in the source rather than in the CMake configuration?
> 
> There seems to be strong precedent to do it via ifdefs. Take for instance the platform specific malloc files in asan:
> 
> https://github.com/llvm/llvm-project/blob/54c93aabec965469fe7db1f4391a190e3d640feb/compiler-rt/lib/asan/asan_malloc_mac.cpp#L15
> 
> https://github.com/llvm/llvm-project/blob/54c93aabec965469fe7db1f4391a190e3d640feb/compiler-rt/lib/asan/asan_malloc_win.cpp#L16
> 
> https://github.com/llvm/llvm-project/blob/54c93aabec965469fe7db1f4391a190e3d640feb/compiler-rt/lib/asan/asan_malloc_linux.cpp#L17
> 
> I'm not opposed to doing this in CMake (and it would be my first gut reaction in a project I controlled) but this seems to be the way I've seen around the codebase.

OK cool - thanks for summarising these. I'll leave it to others with more experience in the codebase to weigh in here then :)

File: compiler-rt/lib/rtsan/rtsan_interceptors_posix.cpp [hunk 1]
```diff
@@ -8,12 +8,14 @@
 //
 //===----------------------------------------------------------------------===//
 
+#include "sanitizer_common/sanitizer_platform.h"
```
Comment (fmayer) [inline compiler-rt/lib/rtsan/rtsan_interceptors_posix.cpp:11]: why is this include outside of the if?

Comment (cjappl) [reply compiler-rt/lib/rtsan/rtsan_interceptors_posix.cpp:11]: It is what defines SANITIZER_POSIX

**Review by fmayer — APPROVED** [review]

**Pull request merged**